File: rb/features.rb
# Opensanctum SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module OpensanctumFeatures
  def self.make_feature(name)
    case name
    when "base"
      OpensanctumBaseFeature.new
    when "ratelimit"
      OpensanctumRatelimitFeature.new
    when "retry"
      OpensanctumRetryFeature.new
    when "test"
      OpensanctumTestFeature.new
    when "timeout"
      OpensanctumTimeoutFeature.new
    else
      OpensanctumBaseFeature.new
    end
  end
end

File: py/features.py
# Opensanctum SDK feature factory

from feature.base_feature import OpensanctumBaseFeature
from feature.test_feature import OpensanctumTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OpensanctumBaseFeature(),
        "test": lambda: OpensanctumTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
